### ISSUE TITLE
SSM Parameter shorthand incorrect syntax not detected

### ISSUE THREAD

**Issue by gbudge** [aws-cloudformation/cfn-lint#4169]
### CloudFormation Lint Version

cfn-lint 1.32.3

### What operating system are you using?

Mac

### Describe the bug

The incorrect whitespace syntax for short-hand SSM Parameter resolution is not detected by cfn-lint, but is detected by CloudFormation during stack update.

Incorrect: `"{{ resolve:ssm:SomeParameter }}"   # missed by cfn-lint, caught by CloudFormation`

Corrected: `"{{resolve:ssm:SomeParameter}}"`

### Expected behavior

**Minimally Expected**
cfn-lint should fail when whitespace syntax is used: `"{{ resolve:ssm:SomeParameter }}"`

**Bonus Points**
cfn-lint also fails when the resolve:ssm syntax is inverted:

- `"{{ssm:resolve:SomeParameter}}"`
- `"{{ ssm:resolve:SomeParameter }}"`

I am surprised how often I see this happen.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Example of missing incorrect shorthand syntax for SSM parameters.'

Resources:
  MyS3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: "{{ resolve:ssm:SomeParameter }}"
      Tags:
        - Key: CorrectSyntax
          Value: "{{resolve:ssm:SomeParameter}}" # This is correct shorthand syntax
        - Key: IncorrectSyntax
          Value: "{{ resolve:ssm:SomeParameter }}"  # This is incorrect shorthand syntax
```
